### PR TITLE
Add linting to js and jsx files

### PR DESCRIPTION
**Pull request by klingerf**
Updates the webpack config to include eslint as part of the js/jsx module bundling. The set of linting rules is configurable going forward. I picked some that I thought made sense, and updated our javascript files accordingly.

### CODE REVIEW

File: web/app/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "ecmaFeatures": {
+      "jsx": true
+    }
+  },
+  "env": {
+    "browser": true,
+    "es6": true
+  },
+  "extends" : ["eslint:recommended", "plugin:react/recommended"],
+  "rules": {
```
Comment (rmars) [inline web/app/.eslintrc]: I'd also like to add `"indent": ["error", 2]` to the rules at some point. It does change a lot of files, so I understand if you want to hold off on including it in this branch. I'm sure there's a bunch of other rules we can add after this is set up too!

File: web/app/js/components/BarChart.jsx [hunk 2]
```diff
@@ -64,23 +39,38 @@ export default class LineGraph extends React.Component {
...
     this.updateScales();
     this.renderGraph();
   }
 
-  chartData() {
-    let data = this.props.data;
-    let percentages = _.each(data, d => {
-      let p = new Percentage(d.rollup.requestRate, d.rollup.totalRequests);
-      d.shareOfRequests = p.get();
-      d.pretty = p.prettyRate();
```
[the remark this answers is not in the document]
Comment (klingerf) [reply web/app/js/components/BarChart.jsx]: Whoa, good catch. Not sure how I messed that up so badly.

File: web/app/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "arrow-parens": [1, "as-needed"],
+    "comma-dangle": [0], // babel removes dangling commas for us
+    "eqeqeq": [2, "smart"],
+    "indent": [2, 2],
+    "jsx-quotes": 1,
+    "no-console": [1, { allow: ["warn", "error"] }], // minor offense
```
Comment (rmars) [inline web/app/.eslintrc]: nit: for consistency, we should double quote all the property names in this file

File: web/app/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "react/no-did-mount-set-state": 1,
+    "react/no-did-update-set-state": 1,
+    "react/no-multi-comp": [1, { "ignoreStateless": true }],
+    "react/prefer-es6-class": 1,
+    "react/prop-types": 0, // TODO: enable prop type validation
+    "react/self-closing-comp": 1,
+    "react/sort-comp": 1,
+    "react/sort-prop-types": 1,
+    "semi": [2, "always"],
+    "sort-imports": [1, { ignoreCase: true, memberSyntaxSortOrder: ["single", "multiple", "all", "none"] }],
```
Comment (rmars) [inline web/app/.eslintrc]: nit: quoting property names

**Pull request merged**